app/helpers/simulator_helper.rb

```diff
@@ -1,2 +1,25 @@
 module SimulatorHelper
+  def return_image_file(data_url)
+    str = data_url["data:image/jpeg;base64,".length .. -1]
+    if str.to_s.empty?
+      path = Rails.root.join("app/assets/images/empty_project/no_preview.png")
+      image_file = File.open(path, "rb")
+
+    else
+      jpeg       = Base64.decode64(str)
+      image_file = File.new("preview_#{Time.now()}.jpeg", "wb")
+      image_file.write(jpeg)
+    end
+
+    image_file
+  end
+
+  def check_to_delete (data_url)
+    str = data_url["data:image/jpeg;base64,".length .. -1]
+    if str.to_s.empty?
```

Comment:
This is same as `!str.to_s.empty?`

Reply:
But we are not declaring str in controller . So we can't use it directly

Reply:
That's fine, I was suggesting to change the implementation of the helper. :sweat_smile: to a one liner say - `data_url["data:image/jpeg;base64,".length .. -1].to_s.empty?`
Plus `File.delete(image_url) if data_url["data:image/jpeg;base64,".length .. -1].to_s.empty?` seems pretty readable.

Reply:
Bhai data_url bhi define nahi kiya controller mein.

Reply:
Does this work? 
```
def check_to_delete (data_url)
    !data_url["data:image/jpeg;base64,".length .. -1.to_s.empty
end
```

Reply:
yup this would work :). but i prefer to do it this way .

Reply:
Okay :+1: , I was suggesting for a one-line implementation for helper. Pardon me for not being more clear here. :smile:

Reply:
I agree with ankit here. This is a standard ruby idiom

Reply:
cool